src/Tests/Microsoft.NET.Build.Tests/GivenWeWantToRequireWindowsForDesktopApps.cs

```diff
@@ -0,0 +1,141 @@
+        [InlineData("Microsoft.WindowsDesktop.App.WPF")]
+        public void It_builds_on_windows_with_a_framework_reference(string desktopFramework)
+        {
+            const string ProjectName = "WindowsDesktopReferenceTest";
+
+            var asset = CreateWindowsDesktopReferenceTestAsset(ProjectName, desktopFramework);
+            
+            var command = new BuildCommand(Log, Path.Combine(asset.Path, ProjectName));
+
+            command
+                .Execute("/restore")
+                .Should()
+                .Pass();
+        }
+
+        [PlatformSpecificTheory(Platform.Linux, Platform.Darwin, Platform.FreeBSD)]
+        [InlineData("Microsoft.WindowsDesktop.App")]
+        [InlineData("Microsoft.WindowsDesktop.App.WindowsForms")]
+        [InlineData("Microsoft.WindowsDesktop.App.WPF")]
+        public void It_errors_on_nonwindows_with_a_framework_reference(string desktopFramework)
+        {
+            const string ProjectName = "WindowsDesktopReferenceErrorTest";
+
+            var asset = CreateWindowsDesktopReferenceTestAsset(ProjectName, desktopFramework);
+            
+            var command = new BuildCommand(Log, Path.Combine(asset.Path, ProjectName));
+
+            command
+                .Execute("/restore")
+                .Should()
+                .Fail()
+                .And
+                .HaveStdOutContaining(Strings.WindowsDesktopFrameworkRequiresWindows);
+        }
+
+        private TestAsset CreateWindowsDesktopSdkTestAsset(string projectName, string desktopProperty)
+        {
+            const string tfm = "netcoreapp3.0";
+
+            var testProject = new TestProject()
+            {
+                Name = projectName,
+                TargetFrameworks = tfm,
+                IsSdkProject = true,
+                IsWinExe = true,
+            };
+
+            testProject.AdditionalProperties.Add(desktopProperty, "true");
+
+            return _testAssetsManager.CreateTestProject(testProject)
+                .WithProjectChanges(
+                    (filename, project) =>
+                    {
+                        project.Root.Attribute("Sdk").Value = "Microsoft.NET.Sdk.WindowsDesktop";
+                    });
+        }
+
+        private TestAsset CreateWindowsDesktopReferenceTestAsset(string projectName, string desktopFramework)
+        {
+            const string tfm = "netcoreapp3.0";
+
+            var testProject = new TestProject()
+            {
+                Name = projectName,
+                TargetFrameworks = tfm,
+                IsSdkProject = true,
+                IsWinExe = true,
+            };
+
+            return _testAssetsManager.CreateTestProject(testProject)
+                .WithProjectChanges(
+                    (filename, project) =>
+                    {
+                        project.Root.Add(XElement.Parse($@"<ItemGroup><FrameworkReference Include=""{desktopFramework}"" /></ItemGroup>"));
```

Comment:
There's now a FrameworkReferences property on TestProject which you can use instead.

Reply:
Thanks, I wasn't aware of that.  I'll fix.